odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1646,7 +1646,13 @@ def _validate_attributes(self, node, name_manager, node_info):
                 self._validate_classes(node, expr)
 
             elif attr == 'context':
-                vnames = get_expression_field_names(expr) - {'id'}
+                try:
+                    vnames = get_expression_field_names(expr) - {'id'}
+                except SyntaxError as e:
+                    message = _(
+                        'Invalid %(attr)s: %(expr)s \n\n %(e)s', attr=attr, expr=expr, e=e
+                        )
```

Comment:
This is much better, thanks. I think it is always better to explain what we expect:
 `_('Invalid context: %(expr)r is not a valid Python expression \n\n %(e)s', expr=expr, e=e)`

Reply:
Hello @ryv-odoo, changes are done